core/src/com/unciv/logic/automation/unit/CityLocationTileRanker.kt

```diff
@@ -0,0 +1,135 @@
+                            .filter { it.isExplored(unit.civ) || unit.civ.isAI() }
+                            .filter { it.getContinent() == center.getContinent() }
+                    )
+                    yieldAll(
+                        center.getTilesInDistance(modConstants.minimalCityDistanceOnDifferentContinents)
+                            .filter { it.isExplored(unit.civ) || unit.civ.isAI() }
+                            .filter { it.getContinent() != center.getContinent() }
+                    )
+                }
+            }.toSet()
+
+            // This is to improve performance - instead of ranking each tile in the area up to 19 times, do it once.
+            val nearbyTileRankings = getNearbyTileRankings(unit.getTile(), unit.civ)
+
+            val distanceFromHome = if (unit.civ.cities.isEmpty()) 0
+            else unit.civ.cities.minOf { it.getCenterTile().aerialDistanceTo(unit.getTile()) }
+            val range = max(
+                1,
+                min(5, 8 - distanceFromHome)
+            ) // Restrict vision when far from home to avoid death marches
+
+            val possibleCityLocations = unit.getTile().getTilesInDistance(range)
+                .filter { it.isExplored(unit.civ) || unit.civ.isAI() }
+                .filter {
+                    val tileOwner = it.getOwner()
+                    it.isLand && !it.isImpassible() && (tileOwner == null || tileOwner == unit.civ) // don't allow settler to settle inside other civ's territory
+                            && (unit.currentTile == it || unit.movement.canMoveTo(it))
+                            && it !in tilesNearCities
+                }.toList()
+
+            val luxuryResourcesInCivArea = getLuxuryResourcesInCivArea(unit.civ)
+
+            return possibleCityLocations
+                .map {
+                    Pair(
+                        it,
+                        rankTileAsCityCenterWithCachedValues(
+                            it,
+                            nearbyTileRankings,
+                            luxuryResourcesInCivArea,
+                            unit.civ
+                        ),
+                    )
+                }
+                .sortedByDescending { it.second }
+        }
+
+        fun rankTileAsCityCenter(tile: Tile, civ: Civilization): Float {
+            val nearbyTileRankings = getNearbyTileRankings(tile, civ)
+            val luxuryResourcesInCivArea = getLuxuryResourcesInCivArea(civ)
+            return rankTileAsCityCenterWithCachedValues(
+                tile,
+                nearbyTileRankings,
+                luxuryResourcesInCivArea,
+                civ
+            )
+        }
+
+        private fun getNearbyTileRankings(
+            tile: Tile,
+            civ: Civilization
+        ): Map<Tile, Float> {
+            return tile.getTilesInDistance(7)
+                .filter { it.isExplored(civ) || civ.isAI() }
+                .associateBy({ it }, { Automation.rankTile(it, civ) })
```

Comment:
We should already have the tile rankings at the top, as nearbyTileRankings, for performance reasons (we don't want to have to calculate them for each tile we're considering as city center)